qutip/models/dicke.py

```diff
@@ -0,0 +1,1325 @@
+                    jp_operator[s,s+1] = ap(j,m-1)
+                if (s - 1) in range(0,nds):
+                    jm_operator[s,s-1] = am(j,m+1)
+                s = s + 1
+    jx_operator = 1/2*(jp_operator + jm_operator)
+    jy_operator = 1j/2*(jm_operator - jp_operator)
+
+    j_alg = [Qobj(jx_operator), Qobj(jy_operator), Qobj(jz_operator), Qobj(jp_operator), Qobj(jm_operator)]
+
+    return j_alg
+
+
+def c_ops_tls(N=2, emission=0., dephasing=0., pumping=0., collective_emission=0.,
+              collective_dephasing=0., collective_pumping=0.):
+    """
+    Create the collapse operators (c_ops) of the Lindblad master equation in
+    the in the uncoupled basis of the two-level system (TLS) SU(2) Pauli matrices. 
+    The collapse operator list can be given to the Qutip algorithm 'mesolve'. 
+    Notice that the operators are placed in a Hilbert space of dimension 2^N.
+    Thus the method is suitable only for small N, N max of order 10.
+
+    Parameters
+    ----------
+    N: int
+        The number of two level systems
+        default = 2
+    emission: float
+        default = 0
+        incoherent emission coefficient (also nonradiative emission)
+    dephasing: float
+        default = 0
+        Dephasing coefficient
+    pumping: float
+        default = 0
+        Incoherent pumping coefficient
+    collective_emission: float
+        default = 0
+        Collective (superradiant) emission coefficient
+    collective_dephasing: float
+        default = 0
+        Collective dephasing coefficient
+    collective_pumping: float
+        default = 0
+        Collective pumping coefficient
+
+    Returns
+    -------
+    c_ops: c_ops vector of matrices
+        c_ops contains the collapse operators for the Lindbla
+
+    """
+    N = int(N)
+
+    if N > 10:
+        print("""Warning! N > 10. dim(H) = 2^N. Use the permutational
+              invariant methods for large N. """)
+
+    [sx, sy, sz, sm, sp] = su2_algebra(N)
+    [jx, jy, jz, jm, jp] = collective_algebra(N)
+
+    c_ops = []
+
+    if emission != 0:
+        for i in range(0, N):
+            c_ops.append(np.sqrt(emission) * sm[i])
+
+    if dephasing != 0:
+        for i in range(0, N):
+            c_ops.append(np.sqrt(dephasing) * sz[i])
+
+    if pumping != 0:
+        for i in range(0, N):
+            c_ops.append(np.sqrt(pumping) * sp[i])
+
+    if collective_emission != 0:
+        c_ops.append(np.sqrt(collective_emission) * jm)
+
+    if collective_dephasing != 0:
+        c_ops.append(np.sqrt(collective_dephasing) * jz)
+
+    if collective_pumping != 0:
+        c_ops.append(np.sqrt(collective_pumping) * jp)
+
+    return c_ops
+
+# ============================================================================
+# State definitions in the Dicke basis with an option for basis transformation
+# ============================================================================
+def dicke_basis(N, jmm1 = None, basis = "dicke"):
```

Comment:
pep8: no spaces in kwargs `jmm1 = None` should be `jmm1=None`. 
There are lots of examples in this file. I'll not mark them all